### PR TITLE
Update Meter component to hooks

### PR DESCRIPTION
**Pull request by sdowding**
#### What does this PR do?
Update the Meter component index to hooks and change getDerivedStateFromProps to use memo. Now excessive rerenders should be prevented, and state is removed. (Party?) 

also, your tests were very useful in completing this tasks.

#### Where should the reviewer start?
Do u think useMemo is required even? i don't have context as to why the original getDerivedStateFromProps was added.

#### What testing has been done on this PR?
I ran it in storybook and the unit tests 💚 passed.

#### How should this be manually tested?
Running the story in storybook would be wise?
http://localhost:9001/?path=/story/meter--bar
http://localhost:9001/?path=/story/meter--circle
http://localhost:9001/?path=/story/meter--multiple-values

#### Any background context you want to provide?
no, im just trying to help out and learn hooks.

#### What are the relevant issues?
#3394

#### Screenshots (if appropriate)
nope

#### Do the grommet docs need to be updated?
no

#### Should this PR be mentioned in the release notes?
no

#### Is this change backwards compatible or is it a breaking change?
backwards compatible.

### CODE REVIEW

File: src/js/components/Meter/Meter.js [hunk 1]
```diff
@@ -14,38 +14,23 @@ const deriveMax = values => {
...
-      content = <Circle max={max} {...rest} />;
-    }
-
-    return content;
-  }
+Meter.defaultProps = {
+  background: { color: 'light-2', opacity: 'medium' },
+  size: 'medium',
+  thickness: 'medium',
+  type: 'bar',
```
[the remark this answers is not in the document]
Comment (sdowding) [reply src/js/components/Meter/Meter.js]: I'm not sure what you mean... defaultProps seem to be the pattern use in the other functional components.

[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/js/components/Meter/Meter.js]: we are trying to move away from initializing props on defaultProps, see for example https://github.com/grommet/grommet/blob/5f651b28de8fae8ac08cdf68b5d085edff52b1c5/src/js/components/Accordion/Accordion.js#L10, the initialization is happening as part of the props declaration.

**Comment by sdowding** [comment]
i have removed the default props, let me know what u think @ShimiSun

File: src/js/components/Meter/Meter.js [hunk 2]
```diff
@@ -14,38 +14,23 @@ const deriveMax = values => {
...
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const { max } = prevState;
-    const nextMax = deriveMax(nextProps.values);
-    if (!max || nextMax !== max) {
-      return { max: nextMax };
-    }
-    return null;
+const Meter = (props) =>  {
+  const { type, ...rest } = props;
```
[the remark this answers is not in the document]
Comment (sdowding) [reply src/js/components/Meter/Meter.js]: done

**Pull request merged**